code/clue.js

```diff
@@ -1,59 +1,353 @@
+}
+
+const axe = {
+  name: 'Axe',
+  weight: 8,
+  id: 'axe'
+}
+
+const bat = {
+  name: 'Bat',
+  weight: 5,
+  id: 'bat'
+}
+
+const trophy = {
+  name: 'Trophy',
+  weight: 5,
+  id: 'trophy'
+}
+
+const pistol = {
+  name: 'Pistol',
+  weight: 4,
+  id: 'pistol'
+}
+
+
+//Grouped into arrays
 
 const suspects = [
   mrGreen,
-  mrsWhite
-  // ...  and the rest
+  mrsWhite,
+  professorPlum,
+  missScarlet,
+  mrsPeacock,
+  colonelMustard,
+]
+
+const weapons = [
+  rope,
+  knife,
+  candlestick,
+  dumbbell,
+  poison,
+  axe,
+  bat,
+  trophy,
+  pistol
+]
+
+
+const rooms = [
+  'Dining Room',
+  'Conservatory',
+  'Kitchen',
+  'Study',
+  'Library',
+  'Billiard Room',
+  'Lounge',
+  'Ballroom',
+  'Hall',
+  'Spa',
+  'Living Room',
+  'Observatory',
+  'Theater',
+  'Guest House',
+  'Patio'
 ]
 
-const weapons = []
+let mystery = {
+  killer: 'who',
+  weapon: 'what',
+  room: 'where'
+}
+
+let killerWeapons = Array(8).fill('');
+let moreWeapons = []; 
 
-const rooms = []
+// FUNCTIONS
 
 // THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
 const randomSelector = array => {
   return array[Math.floor(Math.random() * array.length)]
 }
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
 
-// This function will be invoked when you click on the killer card.
-const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
+const shuffleFavouriteWeapon = () => {
+  suspects.forEach(element => {
+    element.favouriteWeapon = randomSelector(weapons)
+  });
+  
+}
+  
+const showKillerLoader = () => {
+  document.getElementById('killerLoader').style.display = 'flex';
+  document.getElementById('killerHeadline').style.display = 'none';
+  document.getElementById('killerTitle').style.display = 'none';
+  document.getElementById('killerContent').style.display = 'none';
+  document.getElementById('killerDecoration').style.display = 'flex'
+  document.getElementById('killerCard').style.background = 'black'
+  document.getElementById('killerDecoration').style.border = '8px solid rgb(129, 164, 175)'
+}
+
+const showWeaponLoader = () => {
+  document.getElementById('weaponLoader').style.display = 'flex';
+  document.getElementById('weaponRoundLoader').classList.add('weapon-loader');
+  document.getElementById('weaponSmallLoader').style.borderRight = '5px solid #575A53';
+  document.getElementById('weaponLoaderStick').style.backgroundColor = '#575A53';
+  document.getElementById('weaponHeadline').style.display = 'none';
+  document.getElementById('weaponTitle').style.display = 'none';
+  document.getElementById('weaponContent').style.display = 'none';
+  document.getElementById('weaponDecoration').style.display = 'flex'
+  document.getElementById('weaponCard').style.background = 'black'
```

Comment:
You acesses for example the 'weaponCard' multiply times, what you could do is save that in a variable and get cleaner code. 

Like this: `const weaponCard = document.getElementById('weaponCard');`

Reply:
Good suggestion, look for other places you could do this as well